app/models/voucher.rb

```diff
@@ -10,12 +10,14 @@ class Voucher < ApplicationRecord
            class_name: 'Spree::Adjustment',
            dependent: :nullify
 
-  before_validation :strip_code
-
   validates :code, length: { maximum: STRING_COLUMN_LIMIT },
                    presence: true, uniqueness: { scope: :enterprise_id }
   validates :amount, presence: true, numericality: { greater_than: 0 }
 
+  def code=(value)
+    super(value.to_s.strip)
+  end
```

Comment:
My version contained the `&.` operator in case we are assigning `nil`. That is unlikely, but we may do that in specs for example to test validations. It's always good to have robust code that doesn't raise errors on nil values.

Reply:
Oh, I just realised that `to_s` is converting `nil` to a string already and there since the empty string is as invalid as nil, this works fine. :heavy_check_mark: